app/component/ItineraryPage.js

```diff
@@ -607,361 +378,33 @@ class SummaryPage extends React.Component {
 
   paramsOrQueryHaveChanged = () => {
     return (
-      !isEqual(this.params, this.context.match.params) ||
-      !isEqual(this.query, this.context.match.location.query)
+      !isEqual(this.params, this.props.match.params) ||
```

Comment:
match is usually loaded through context, I don't remember anymore why we sometimes load it from props.

Reply:
I think there was more props.match accees than context.match so I took the more popular. Match is available in props because router automatically puts it there. I verified that match.params are the same in both cases.